src/components/data-exploration/cell-sets-tool/CellSetsTool.jsx

```diff
@@ -231,6 +228,18 @@ const CellSetsTool = (props) => {
     );
   };
 
+  if (loading) return <Skeleton active />;
+  if (!cellSetTreeData || !metadataTreeData) return <Skeleton active />;
```

Comment:
why not in the same if statement? Also, why putting a skeleton when there is no cell set or metadata data? Wouldn't that confuse the user that something is loading, when actually nothing is getting loaded?

Reply:
There is not really a reason for not putting them together, but I have found that creating one big `if` statement can make things harder to read. In addition, in the flow, `loading` will be first set to `false`, and **then** we'll have to wait for `cellSetTreeData` and `metadataTreeData` to be defined, so this better reflects the actual steps that go into loading the cell sets imho.

In addition, `cellSetTreeData` and `metadataTreeData` are never `null` in their normal state. They may be an empty array, but that's not a falsy value in Javascript. The only time these values are falsy is when the hook to create `cellSetTreeData` and `metadataTreeData` has not been run yet.

Reply:
In terms of the loading explanation: sounds like it is **very important** that loading states are always set to false initially by default, because if we set them to true, this will make a component never load. I am a bit concerned that it is very easy to not think about this and create a new loading state and set it to true by default (I actually saw it in a PR a few months ago). So what can we do to make sure that this never happens?

Reply:
Don't see why that would be the case. As it is right now, the component loads if `loading` is `false`, and `cellSetTreeData` is truthy, and `metadataTreeData` is truthy. You can set `loading` to be `true` by default, there's nothing wrong with it, the component will not load until `loading` is `false`.